### PR TITLE
Add user de/activation and emails

### LINKED ISSUE #178: "New query" button on hospital page

### LINKED ISSUE #282: Hinweis: Mails von Krankenhäusern werden erst nach Validierung herausgeschickt
Hinweis-Popup nach dem Absenden der Mails (bei Neuregistrierung):
"Wir speichern ihre Mails und senden sie sobald sie als Krankenhaus validiert sind"

### LINKED ISSUE #297: Email an Studierende
Endpoint, den man als Admin rufen kann
- [x] Link unten in Krankennausemail zum Account deaktivieren
- [ ] Sucht alle user, die vor mehr als 3 Tagen eine email bekommen haben, denen schicken wir eine email "Hast du dich gematcht? wenn ja unsubscibe"
- [x] Limit Emails pro krankenhaus pro tag. "If you have a good reason to send more emails, wendet euch an support@match4healthcare.de"

solve #231 while at it

### PR DESCRIPTION
**Pull request by feeds**
Should close
Closes #178
Closes #297
Closes #282

### CODE REVIEW

File: backend/apps/accounts/templates/change_activation_ask.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{% extends 'base.html' %}
+{% load static %}
+{% load i18n %}
+
+{% block title %}
+Account unwiderruflich löschen?
```
Comment (bjrne) [inline backend/apps/accounts/templates/change_activation_ask.html]: Translation Block
```suggestion
{% blocktrans %} Account unwiderruflich löschen?{% endblocktrans %}
```

File: backend/apps/accounts/templates/change_activation_ask.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ <div class="col-lg-6 col-md-10 col-sm-12 col-12" >
+  <div class="card login-card text-center align-items-center">
+      {% if is_activated %}
+      <h4>{% blocktrans %}
+          Bist du sicher, dass du deinen Account vorerst deaktivieren möchtest?{% endblocktrans %}</h4>
+          <p>{% blocktrans %}Dann kannst du nicht mehr von Hilfesuchenden kontaktiert werden.
+          Du kannst die Deaktivierung aber jederzeit rückgängig machen. {% endblocktrans %}</p>
+      <a href="change_activation_confirm"><button class="btn blue text-white btn-md" style="margin-top:10px;">{% blocktrans %}Ja, Account deaktivieren.{% endblocktrans %}</button></a>
+    {% else %}
+       <h4>{% blocktrans %}
```
Comment (bjrne) [inline backend/apps/accounts/templates/change_activation_ask.html]: unnecessary break, harder for translation file, move trans block to next line

File: backend/apps/accounts/views.py [hunk 3]
```diff
@@ -267,3 +268,29 @@ def get(self, request, format=None):
 
 class CustomLoginView(LoginView):
     authentication_form = CustomAuthenticationForm
+
+from django.contrib import messages
```
Comment (bjrne) [inline backend/apps/accounts/views.py]: WHY here :sweat:

File: backend/apps/accounts/views.py [hunk 4]
```diff
@@ -267,3 +268,29 @@ def get(self, request, format=None):
...
+    s = request.user.student
+    status = s.is_activated
+    s.is_activated = not s.is_activated
+    s.save()
+    if status:
+        messages.add_message(request, messages.INFO, _(
+            'Du hast dein Profil erfolreich deaktiviert, du kannst nun keine Anfragen mehr von Hilfesuchenden bekommen.'))
+    else:
+        messages.add_message(request, messages.INFO,_(
+            'Du hast dein Profil erfolreich aktiviert, du kannst nun wieder von Hilfesuchenden kontaktiert werden.'))
```
Comment (bjrne) [inline backend/apps/accounts/views.py]: typo erfolreich

File: backend/apps/accounts/views.py [hunk 5]
```diff
@@ -267,3 +268,29 @@ def get(self, request, format=None):
...
+@login_required
+@student_required
+def change_activation(request):
+    s = request.user.student
+    status = s.is_activated
+    s.is_activated = not s.is_activated
+    s.save()
+    if status:
+        messages.add_message(request, messages.INFO, _(
+            'Du hast dein Profil erfolreich deaktiviert, du kannst nun keine Anfragen mehr von Hilfesuchenden bekommen.'))
```
Comment (bjrne) [inline backend/apps/accounts/views.py]: same typo

File: backend/apps/iamstudent/templates/send_mail_hospital.html [hunk 6]
```diff
@@ -27,16 +27,17 @@
             <h4>{% blocktrans %}Anzeige versenden{% endblocktrans %}</h4>
         </div>
         <div class="card-body">
-            <form action="/iamstudent/send_mail_student/{{ids}}" method="post">
+            <form action="/iamstudent/send_mail_student/{{ids}}" method="post" onsubmit="disableButton()">
                 {% csrf_token %}
                 {{ form | crispy }}
                 <p>
-       <button type="submit" class="btn btn-success">Sende Mails</button>
+       <button type="submit" class="btn btn-success" id="btSubmitForm">Sende Mails</button>
```
Comment (bjrne) [inline backend/apps/iamstudent/templates/send_mail_hospital.html]: Translation

File: backend/apps/iamstudent/views.py [hunk 7]
```diff
@@ -49,34 +47,39 @@ def get_student(request):
...
+    if max_emails < len(id_list):
+        pass
+        # do something
+
     if request.method == 'POST':
         # create a form instance and populate it with data from the request:
         form = EmailToSendForm(request.POST)
 
         if form.is_valid():
-            # todo check header injections!!!!
```
Comment (bjrne) [inline backend/apps/iamstudent/views.py:70]: Do we?

File: backend/apps/iamstudent/views.py [hunk 8]
```diff
@@ -49,34 +47,39 @@ def get_student(request):
...
-
 
+            hospital_message = form.cleaned_data['message']
 
             subject = form.cleaned_data['subject']
             for student_id in id_list:
                 student = Student.objects.get(user_id=student_id)
 
-                message = 'Hallo %s %s,\n\n wir haben folgende Nachricht von %s für dich.\n\nDein Match4MedisTeam\n\n%s' % (
+                message = 'Hallo %s %s,\n\n wir haben folgende Nachricht von %s für dich. Falls du keine Nachrichten mehr erhalten möchtest deaktiviere dein Konto hier: https://match4healthcare.de/accounts/change_activation\n\nDein Match4Healthcare Team\n\n%s' % (
```
Comment (bjrne) [inline backend/apps/iamstudent/views.py:82]: translation

**Review by bjrne — COMMENTED** [review]
more work for you :bee:

File: backend/apps/iamstudent/views.py (hunk 7 above)
Comment (feeds) [reply backend/apps/iamstudent/views.py:70]: yes, jonny tested this ages ago, its a feature of django itself

File: backend/apps/iamstudent/views.py (hunk 8 above)
Comment (bjrne) [reply backend/apps/iamstudent/views.py:82]: left for #312, needs testing for proper format handling

**Review by bjrne — APPROVED** [review]
Now approved

**Pull request merged**